### DISCUSSION TITLE
The calculation results do not appear.

### DISCUSSION THREAD

**Discussion by kdm0503** [community/community#102021] [Programming Help] [answered]
### Body

A window to input AAD and BB will appear, but no results will appear. I would really appreciate it if you could tell me how to solve it.

--------------------------------------------------------------------------------------------------------------------------------------
<!DOCTYPE html>
<html lang="ko-KO">
<body>

    <div class="nav">
        <ul>
            <li><a href="#" onclick="showSection('home')">홈</a></li>
            <li><a href="#" onclick="showSection('measurementSection')">어깨너비</a></li>
            <li><a href="#" onclick="showSection('SBDSection')">중량 기록</a></li>
            <li><a href="#" onclick="showSection('DeltaSection')">변화 </a></li>
        </ul>
    </div>

    <div id="home"> 
        <h2>시작 화면</h2>
        <p>여기에 원하는 내용을 추가하세요.</p>
    </div>

    <div id="measurementSection" style="display:none;">
        <p>연령과 어깨 골격 너비를 입력하시면 본인의 어깨 골격 너비 백분위를 알 수 있습니다.</p>
        <select id="ageGroupAAD">
            <option value="16">만 16세</option>
            <option value="17">만 17세</option>
            <option value="18">만 18세</option>
            <option value="19">만 19세</option>
            <option value="20 ~ 24">만 20 ~ 24세</option>
            <option value="25 ~ 29">만 25 ~ 29세</option>
            <option value="30 ~ 1000">만 30세 이상</option>
        </select>
        <input type="number" id="AAD" placeholder="어깨 골격 너비를 입력해주세요.">
        <button onclick="calculatePercentileAAD(zScoreAAD, 'resultAAD')">AAD 확인</button>
        <p id="resultAAD"></p>

        <p>연령과 위 팔 사이너비를 입력하시면 본인의 위 팔 사이너비 백분위를 알 수 있습니다.</p>
        <select id="ageGroupBB">
            <option value="16">만 16세</option>
            <option value="17">만 17세</option>
            <option value="18">만 18세</option>
            <option value="19">만 19세</option>
            <option value="20 ~ 24">만 20 ~ 24세</option>
            <option value="25 ~ 29">만 25 ~ 29세</option>
            <option value="30 ~ 39">만 30 ~ 39세</option>
            <option value="40 ~ 44">만 40 ~ 44세</option>
            <option value="45 ~ 49">만 45 ~ 49세</option>
            <option value="50 ~ 59">만 50 ~ 59세</option>
            <option value="60 ~ inf">만 60세 이상</option>
        </select>
        <input type="number" id="BB" placeholder="위 팔 사이너비를 입력해주세요.">
        <button onclick="calculatePercentileBB(zScoreBB, 'resultBB')">BB 확인</button>
        <p id="resultBB"></p>
    </div>

    <script>

        function showSection(sectionId) {
            // 모든 섹션 숨기기
            document.getElementById('home').style.display = 'none';
            document.getElementById('measurementSection').style.display = 'none';

            // 선택된 섹션만 표시
            document.getElementById(sectionId).style.display = 'block';
        }

        // 기존의 JavaScript 함수들

        function calculatePercentileAAD() {
            var ageGroup = document.getElementById('ageGroupAAD').value;
            var AAD = document.getElementById('AAD').value;
            const sAAD = 1.85; // AAD 표준 편차
            let mAAD; // 평균 AAD, 연령대에 따라 달라짐

            // AAD 연령대별 평균값 설정
            switch(ageGroupAAD) {
                case '16':
                    mAAD = 38.6;
                    break;
                case '17':
                    mAAD = 39.0;
                    break;
                case '18':
                    mAAD = 39.3;
                    break;
                case '19':
                    mAAD = 39.6;
                    break;
                case '20 ~ 24':
                    mAAD = 39.9;
                    break;
                case '25 ~ 29':
                    mAAD = 40.1;
                    break;
                case '30 ~ 1000':
                    mAAD = 40.2;
                    break;
                default:
                    mAAD = 0; // 기본값
            }

            let zScoreAAD = (AAD - mAAD) / sAAD;
            zScoreAAD = Math.abs(zScoreAAD); // Z 점수 음수일 경우 양수로 변환

            // AAD 상위 백분위 계산
            calculatePercentile(document.getElementById('zScoreAAD').value, document.getElementById('ageGroupAAD').value, 'resultAAD');
        }

            function calculatePercentileBB() {
            var ageGroup = document.getElementById('ageGroupBB').value;
            var BB = document.getElementById('BB').value;
            const sBB = 2.87; // BB 표준 편차
            let mBB; // 평균 BB, 연령대에 따라 달라짐

            // BB 연령대별 평균값 설정
            switch(ageGroupBB) {
                case '16':
                    mBB = 44.0;
                    break;
                case '17':
                    mBB = 44.7;
                    break;
                case '18':
                    mBB = 45.1;
                    break;
                case '19':
                    mBB = 45.8;
                    break;
                case '20 ~ 24':
                    mBB = 48.8;
                    break;
                case '25 ~ 29':
                    mBB = 49.0;
                    break;
                case '30 ~ 39':
                    mBB = 49.1;
                    break;
                case '40 ~ 44':
                    mBB = 48.5;
                    break;
                case '45 ~ 49':
                    mBB = 48.1;
                    break;
                case '50 ~ 59':
                    mBB = 47.0;
                    break;
                case '60 ~ inf':
                    mBB = 46.0;
                    break;
                default:
                    mBB = 0; // 기본값
            }

            let zScoreBB = (BB - mBB) / sBB;
            zScoreBB = Math.abs(zScoreBB); // Z 점수 음수일 경우 양수로 변환

            // BB 상위 백분위 계산
            calculatePercentile(document.getElementById('zScoreBB').value, document.getElementById('ageGroupBB').value, 'resultBB');
        }
        function calculatePercentile(zScore, resultAAD, resultBB) {
            fetch('z_table_0_to_10.csv')
                .then(response => response.text())
                .then(data => {
                    const lines = data.split('\n');
                    for (let line of lines) {
                        const [z, percentile] = line.split(',');
                        if (zScore <= parseFloat(z)) {
                            document.getElementById(resultId).innerText = 
                                `상위 ${100 - (parseFloat(percentile) * 100)}% 입니다.`;
                            break;
                        }
                    }
                })
                .catch(error => console.error('Error:', error));
        }
    </script>
</body>
</html>

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by shubhangamjha** [comment]
Well just copy the whole code and try to use the same code with diffrent file name. maybe your file has got corrupted

**Comment by sbk2k1** [comment, marked as answer]
Hey @kdm0503 if I understand correctly, you can't populate the calculated results correctly. Here are the steps you can use:

- Remove the parameters from onclick fuctions
```
<button onclick="calculatePercentileAAD()">Check AAD</button>
```
and
``` 
<button onclick="calculatePercentileBB()">Check BB</button>
```
- set innerHTML to the value calculated
```
// populate zScoreBB
     document.getElementById('resultBB').innerHTML = zScoreBB;
```
and
```
// populate zScoreAAD
    document.getElementById('resultAAD').innerHTML = zScoreAAD;
```

---

Entire Code translated to english:
```
<div class="nav">
 <ul>
     <li><a href="#" onclick="showSection('home')">Home</a></li>
     <li><a href="#" onclick="showSection('measurementSection')">Shoulder Width</a></li>
     <li><a href="#" onclick="showSection('SBDSection')">Weight Records</a></li>
     <li><a href="#" onclick="showSection('DeltaSection')">Changes</a></li>
 </ul>
</div>

<div id="home">
 <h2>Home Screen</h2>
 <p>Add your desired content here.</p>
</div>

<div id="measurementSection" style="display:none;">
 <p>Enter your age and shoulder skeletal width to find your shoulder skeletal width percentile.</p>
 <select id="ageGroupAAD">
     <option value="16">Age 16</option>
     <option value="17">Age 17</option>
     <option value="18">Age 18</option>
     <option value="19">Age 19</option>
     <option value="20 ~ 24">Age 20-24</option>
     <option value="25 ~ 29">Age 25-29</option>
     <option value="30 ~ 1000">Age 30 and above</option>
 </select>
 <input type="number" id="AAD" placeholder="Enter shoulder skeletal width">
 <button onclick="calculatePercentileAAD()">Check AAD</button>
 <p id="resultAAD"></p>

 <p>Enter your age and upper arm span ratio to find your upper arm span ratio percentile.</p>
 <select id="ageGroupBB">
     <option value="16">Age 16</option>
     <option value="17">Age 17</option>
     <option value="18">Age 18</option>
     <option value="19">Age 19</option>
     <option value="20 ~ 24">Age 20-24</option>
     <option value="25 ~ 29">Age 25-29</option>
     <option value="30 ~ 39">Age 30-39</option>
     <option value="40 ~ 44">Age 40-44</option>
     <option value="45 ~ 49">Age 45-49</option>
     <option value="50 ~ 59">Age 50-59</option>
     <option value="60 ~ inf">Age 60 and above</option>
 </select>
 <input type="number" id="BB" placeholder="Enter upper arm span ratio">
 <button onclick="calculatePercentileBB()">Check BB</button>
 <p id="resultBB"></p>
</div>

<script>
 function showSection(sectionId) {
     // Hide all sections
     document.getElementById('home').style.display = 'none';
     document.getElementById('measurementSection').style.display = 'none';

     // Show the selected section
     document.getElementById(sectionId).style.display = 'block';
 }

 // Existing JavaScript functions

 function calculatePercentileAAD() {
     var ageGroup = document.getElementById('ageGroupAAD').value;
     var AAD = document.getElementById('AAD').value;
     const sAAD = 1.85; // AAD standard deviation
     let mAAD; // Average AAD, varies with age group

     // Set AAD average values based on age group
     switch (ageGroupAAD) {
         case '16':
             mAAD = 38.6;
             break;
         case '17':
             mAAD = 39.0;
             break;
         case '18':
             mAAD = 39.3;
             break;
         case '19':
             mAAD = 39.6;
             break;
         case '20 ~ 24':
             mAAD = 39.9;
             break;
         case '25 ~ 29':
             mAAD = 40.1;
             break;
         case '30 ~ 1000':
             mAAD = 40.2;
             break;
         default:
             mAAD = 0; // Default value
     }

     let zScoreAAD = (AAD - mAAD) / sAAD;
     zScoreAAD = Math.abs(zScoreAAD); // Convert Z score to positive if negative

    // populate zScoreAAD
    document.getElementById('resultAAD').innerHTML = zScoreAAD;

     // Calculate AAD upper percentile
     calculatePercentile(document.getElementById('zScoreAAD').value, document.getElementById('ageGroupAAD').value, 'resultAAD');
 }

 function calculatePercentileBB() {
     var ageGroup = document.getElementById('ageGroupBB').value;
     var BB = document.getElementById('BB').value;
     const sBB = 2.87; // BB standard deviation
     let mBB; // Average BB, varies with age group

     // Set BB average values based on age group
     switch (ageGroupBB) {
         case '16':
             mBB = 44.0;
             break;
         case '17':
             mBB = 44.7;
             break;
         case '18':
             mBB = 45.1;
             break;
         case '19':
             mBB = 45.8;
             break;
         case '20 ~ 24':
             mBB = 48.8;
             break;
         case '25 ~ 29':
             mBB = 49.0;
             break;
         case '30 ~ 39':
             mBB = 49.1;
             break;
         case '40 ~ 44':
             mBB = 48.5;
             break;
         case '45 ~ 49':
             mBB = 48.1;
             break;
         case '50 ~ 59':
             mBB = 47.0;
             break;
         case '60 ~ inf':
             mBB = 46.0;
             break;
         default:
             mBB = 0; // Default value
     }

     let zScoreBB = (BB - mBB) / sBB;
     zScoreBB = Math.abs(zScoreBB); // Convert Z score to positive if negative

     // populate zScoreBB
     document.getElementById('resultBB').innerHTML = zScoreBB;

     // Calculate BB upper percentile
     calculatePercentile(document.getElementById('zScoreBB').value, document.getElementById('ageGroupBB').value, 'resultBB');
 }

 function calculatePercentile(zScore, resultAAD, resultBB) {
     fetch('z_table_0_to_10.csv')
         .then(response => response.text())
         .then(data => {
             const lines = data.split('\n');
             for (let line of lines) {
                 const [z, percentile] = line.split(',');
                 if (zScore <= parseFloat(z)) {
                     document.getElementById(resultId).innerText =
                         `Upper ${100 - (parseFloat(percentile) * 100)}%.`;
                     break;
                 }
             }
         })
         .catch(error => console.error('Error:', error));
 }
</script>
```

**Reply by kdm0503** [reply]
WOOW thank you. It prints normally. But the problem is that the calculation results come out strangely. For example, when entering AAD, it should be displayed as Percentile (%) according to Z-score, but a strange rational number appears. I would really appreciate it if you could tell me what to do. very thanks!
![image](https://github.com/community/community/assets/157478342/2e2bf28a-3237-401b-90d1-06083d96ff51)

**Reply by sbk2k1** [reply]
I think that is a problem in the calculation logic. Could you exactly define here, the logic? It would be faster than going through the code.

**Reply by kdm0503** [reply]
(data value - data average) / Standard Deviation = Z-score, Here, it is obtained using the standard normal distribution table rather than using the pure z value, and I used (fetch part z_table_0_to_10.csv). After finding the corresponding value for that z value, you can find out what percentage is at the top or bottom by multiplying (1-corresponding value)*100. However, I am asking this question because the actual calculation result is not in the % form, but rather an incorrect value as shown in the picture.

**Reply by sbk2k1** [reply]
Can you send the .csv file so I can check the app myself?

**Reply by kdm0503** [reply]
The upload is too slow so I'm sending it by email.
